lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -769,6 +773,26 @@ def cmd_irb_help
   def cmd_irb(*args)
     defanged?
 
+    expressions = []
+
+    # Parse the command options
+    @@irb_opts.parse(args) do |opt, idx, val|
+      case opt
+        when '-e'
+          expressions << val
+        when '-h'
+          cmd_irb_help
+          return false
+      end
+    end
+
+    if !expressions.empty?
+      expressions.each do |expression|
+        eval(expression, binding)
+      end
+      return
```

Comment:
I was going to add a ret val here toggled by the return from eval, but eval basically always returns nil and may or may not raise an exception, so I'm not going to do that since I think the user needs the stack trace to figure out what went wrong